### PR TITLE
Added missing allowFontScaling option into tabBarOptions for react-navigation

### PR DESCRIPTION
**Pull request by svbutko**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://reactnavigation.org/docs/navigators/tab#tabBarOptions-for-TabBarBottom-default-tab-bar-on-iOS
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by bang88** [comment]
👍

File: types/react-navigation/index.d.ts [hunk 1]
```diff
@@ -10,8 +10,9 @@
 //                 charlesfamu <https://github.com/charlesfamu>
 //                 Tim Wang <https://github.com/timwangdev>
 //                 Qibang Sun <https://github.com/bang88>
+//                 Sergei Butko: <https://github.com/svbutko>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 2.6
```
Comment (minestarks) [inline types/react-navigation/index.d.ts]: Incrementing this version means you're using features only available in TypeScript version >=2.6, but I don't think this is true for your change. Unless there's something I'm not seeing, please revert this line. Thanks!

**Review by minestarks — CHANGES_REQUESTED** [review]

File: types/react-navigation/index.d.ts (hunk 1 above)
Comment (svbutko) [reply types/react-navigation/index.d.ts]: Sorry, my mistake, reverted this line.

File: types/react-navigation/index.d.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@
 //                 charlesfamu <https://github.com/charlesfamu>
 //                 Tim Wang <https://github.com/timwangdev>
 //                 Qibang Sun <https://github.com/bang88>
+//                 Sergei Butko: <https://github.com/svbutko>
```
Comment (kyler-hyuna) [inline types/react-navigation/index.d.ts]: Also adding your name here, says you agree to be a maintainer for these typings. So you'll be pinged for code reviews and issues. Just making sure you know what you're doing.

**Review by kyler-hyuna — CHANGES_REQUESTED** [review]

File: types/react-navigation/index.d.ts (hunk 2 above)
Comment (svbutko) [reply types/react-navigation/index.d.ts]: Yes, I know about this, just didn't know about TS versioning. Every project depends on this library and I'll try to contribute to it and try to review the code.

Comment (kyler-hyuna) [reply types/react-navigation/index.d.ts]: Great 👍

**Review by kyler-hyuna — APPROVED** [review]

**Review by minestarks — APPROVED** [review]

**Comment by kyler-hyuna** [comment]
Lol @typescript-bot is tired 😆

**Pull request merged**